### ISSUE TITLE
`invalid-typeof`: Allow comparing against identifiers

### ISSUE THREAD

**Issue by marvinhagemeister** [denoland/deno_lint#1375]
Currently this lint rule is complaining about code like this:

```ts
const type = "string";
if (typeof foo === type) {}
```

I feel like that should be accepted by default. TypeScript already checks the type value.
